### PR TITLE
Refactored Storage OpenRead Tests

### PR DESCRIPTION
**Pull request by jaschrep-msft**
Previously, all tests for read streams from OpenRead methods were copy/pasted for each client that could perform the operation. This PR centralizes all those tests into one class and has subclasses provide the client-specific functionality.

### Review Tips
- Start with `sdk/storage/Azure.Storage.Common/tests/Shared/OpenReadTestBase.cs`. Most things will make sense after that.
- If unsure of what patterns to compare against, look at `sdk/storage/Azure.Storage.Common/tests/Shared/TransactionalHashingTestBase.cs`
- `AccessConditionConfigs` centralizes some of the access condition management code that used to be copy/pasted along with these tests. This PR only moves what was necessary for these tests to function into that class (and its subclasses). Filed #25249 to track the rest of that work.

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientOpenReadTests.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        BlobClientOptions.ServiceVersion.V2020_04_08,
+        BlobClientOptions.ServiceVersion.V2020_06_12,
+        BlobClientOptions.ServiceVersion.V2020_08_04,
+        BlobClientOptions.ServiceVersion.V2020_10_02,
+        BlobClientOptions.ServiceVersion.V2020_12_06,
+        StorageVersionExtensions.LatestVersion,
+        StorageVersionExtensions.MaxVersion,
+        RecordingServiceVersion = BlobClientOptions.ServiceVersion.V2020_12_06,
+        LiveServiceVersions = new object[] { StorageVersionExtensions.LatestVersion })]
+    public abstract class BlobBaseClientOpenReadTests<TBlobClient> : OpenReadTestBase<
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientOpenReadTests.cs:33]: Could this class extend `BlobTestBase`, so we could avoid the `ClientTestFixture` above?

Comment (jaschrep-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientOpenReadTests.cs:33]: Unfortunately not. It has to extend OpenReadTestBase, as that is where the tests are actually written and need to be inherited from, and OpenReadTestBase has to work on Blobs, DataLake, and Shares, each of which has a different class that defines service versions.

This unfortunately leaves us with the situation where BlobTestBase, BlobBaseClientTransactionalHashingTests, and now BlobBaseClientOpenReadTests must individually set their ClientTestFixture. I see now that this does become annoying to maintain as this scales out and service versions are updated.

Perhaps the fix is to, in each of the service test packages, define a new attribute that extends ClientTestFixture and supplies all these values there. That way there is once again only one place to update service versions in code per-package, it's just no longer on BlobTestBase or equivalent class. Then all these classes can just be tagged with [BlobsTestFixture] or service name equivalent.

**Review by amnguye — APPROVED** [review]

**Comment by seanmcc-msft** [comment]
Looks like CI is failing over x-ms-version, need to find and replace `"x-ms-version": "2020-12-06"` with `"x-ms-version": "2021-02-12"` in the /tests/SessionRecords folders.

**Review by seanmcc-msft — APPROVED** [review]
Approved pending live tests.

**Pull request merged**